### PR TITLE
feat: add `blas/ext/base/dapxsumkbn-wasm`

### PR DESCRIPTION
**Pull request by Snehil-Shah**
## Description

> What is the purpose of this pull request?

This pull request:

-   adds the package `blas/ext/base/dapxsumkbn-wasm`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   is related to #2039  

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/lib/routine.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+	// Convert the input arrays to "pointers" in the module's memory:
+	ptrs = arrays2ptrs( this, [
+		strided2object( N, x, strideX, offsetX )
+	]);
+	p0 = ptrs[ 0 ];
+
+	// Perform computation by calling the corresponding parent method:
+	return Module.prototype.ndarray.call( this, N, alpha, p0.ptr, p0.stride, p0.offset );
+});
```
Comment (Snehil-Shah) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/lib/routine.js:163]: I can't exactly pinpoint what I'm doing wrong here, but tests are failing for cases when `stride` is zero. The compiled `wasm` functions are working as intended, as the corresponding `Module.prototype` tests are passing.

Comment (saurabhraghuvanshii) [reply lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/lib/routine.js:163]: @Snehil-Shah you should close this PR and reopen it , It can work

Comment (Snehil-Shah) [reply lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/lib/routine.js:163]: @saurabhraghuvanshii not a CI issue, the tests are failing locally as well.

Comment (saurabhraghuvanshii) [reply lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/lib/routine.js:163]: @Snehil-Shah ok , is repo is update to date or rebased?

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/test/test.module.ndarray.js [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+		'initial': 1
+	});
+	mod = new Module( mem );
+	mod.initializeSync();
+
+	xp = 0;
+
+	mod.write( xp, new Float64Array( [ 1.0, -2.0, -4.0, 5.0, 0.0, 3.0 ] ) );
+
+	y = mod.ndarray( 9, 5.0, xp, 1, 0 );
```
Comment (aman-095) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/test/test.module.ndarray.js]: The example has six parameters and hence it should be:
```
y = mod.ndarray( 6, 5.0, xp, 1, 0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/README.md [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var dapxsumkbn = require( '@stdlib/blas/ext/base/dapxsumkbn-wasm' );
+```
+
+#### dapxsumkbn.main( N, alpha, x, stride )
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/README.md]:
```suggestion
#### dapxsumkbn.main( N, alpha, x, strideX )
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/README.md [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+var sum = dapxsumkbn.main( x.length, 5.0, x, 1 );
+// returns 16.0
+```
+
+The function has the following parameters:
+
+-   **N**: number of indexed elements.
+-   **alpha**: scalar constant.
+-   **x**: input [`Float64Array`][@stdlib/array/float64].
+-   **strideX**: index increment for `x`.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/README.md]:
```suggestion
-   **strideX**: stride length for `x`.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/README.md [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+var Float64Array = require( '@stdlib/array/float64' );
+
+var x0 = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0 ] );
+var x1 = new Float64Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // start at 2nd element
+
+var sum = dapxsumkbn.main( 4, 5.0, x1, 2 );
+// returns 25.0
+```
+
+#### dapxsumkbn.ndarray( N, x, strideX, offsetX )
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/README.md]:
```suggestion
#### dapxsumkbn.ndarray( N, alpha, x, strideX, offsetX )
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/README.md [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+
+var sum = dapxsumkbn.ndarray( 3, 5.0, x, 1, 0 );
+// returns 16.0
+```
+
+The function has the following additional parameters:
+
+-   **offsetX**: starting index for `x`.
+
+While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameter supports indexing semantics based on a starting index. For example, to access every other value in `x` starting from the second value
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/README.md]:
```suggestion
While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameter supports indexing semantics based on a starting index. For example, to access every other element starting from the second element:
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt [hunk 7]
```diff
@@ -0,0 +1,514 @@
...
+
+    If `N <= 0`, the function returns `0.0`.
+
+    Parameters
+    ----------
+    N: integer
+        Number of indexed elements.
+
+    alpha: number
+        Constant.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt]:
```suggestion
    alpha: number
        Scalar constant.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt [hunk 8]
```diff
@@ -0,0 +1,514 @@
...
+
+    x: Float64Array
+        Input array.
+
+    strideX: integer
+        Index increment.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt]:
```suggestion
    strideX: integer
        Stride length.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt [hunk 9]
```diff
@@ -0,0 +1,514 @@
...
+        Sum.
+
+    Examples
+    --------
+    // Standard Usage:
+    > var x = new {{alias:@stdlib/array/float64}}( [ 1.0, -2.0, 2.0 ] );
+    > {{alias}}.main( x.length, 5.0, x, 1 )
+    16.0
+
+    // Using `N` and `stride` parameters:
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt]:
```suggestion
    // Using `N` and stride parameters:
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt [hunk 10]
```diff
@@ -0,0 +1,514 @@
...
+    buffer, the offset parameter supports indexing semantics based on a
+    starting index.
+
+    Parameters
+    ----------
+    N: integer
+        Number of indexed elements.
+
+    alpha: number
+        Constant.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt]:
```suggestion
    alpha: number
        Scalar constant.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt [hunk 11]
```diff
@@ -0,0 +1,514 @@
...
+
+    x: Float64Array
+        Input array.
+
+    strideX: integer
+        Index increment.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt]:
```suggestion
    strideX: integer
        Stride length.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt [hunk 12]
```diff
@@ -0,0 +1,514 @@
...
+    array element and computes the sum using an improved Kahan–Babuška
+    algorithm.
+
+    Parameters
+    ----------
+    N: integer
+        Number of indexed elements.
+
+    alpha: number
+        Constant.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt]:
```suggestion
    alpha: number
        Scalar constant.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt [hunk 13]
```diff
@@ -0,0 +1,514 @@
...
+
+    xp: integer
+        Input array pointer (i.e., byte offset).
+
+    sx: integer
+        Index increment.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt]:
```suggestion
    sx: integer
        Stride length.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt [hunk 14]
```diff
@@ -0,0 +1,514 @@
...
+    array element and computes the sum using an improved Kahan–Babuška
+    algorithm and alternative indexing semantics.
+
+    Parameters
+    ----------
+    N: integer
+        Number of indexed elements.
+
+    alpha: number
+        Constant.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt]:
```suggestion
    alpha: number
        Scalar constant.
```

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt [hunk 15]
```diff
@@ -0,0 +1,514 @@
...
+
+    xp: integer
+        Input array pointer (i.e., byte offset).
+
+    sx: integer
+        Index increment.
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/docs/repl.txt]:
```suggestion
    sx: integer
        Stride length.
```

**Review by headlessNode — COMMENTED** [review]
@Snehil-Shah docs related suggestions since we wanna keep consistent terminologies across packages.

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/README.md (hunk 3 above)
Comment (Snehil-Shah) [reply lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/README.md]: I suppose the corresponding `blas/base/dasum-wasm` might also need updation as I followed that as a reference for these terminologies.

Comment (headlessNode) [reply lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/README.md]: Yes it does need updating 😬

File: lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/lib/routine.js (hunk 1 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/lib/routine.js:163]: Hmm...based on CI, I am not sure that the wasm binary corresponds to the upstream C implementation. The test result is actually correct. Namely, when the stride is `0`, the result should actually be `N * ( alpha + x[ offsetX ] )`, not `alpha + x[ offsetX ]`. We've been changing that in the various extended BLAS packages, but it doesn't appear that `dapxsumkbn` has been updated, yet, on `develop`.

Comment (headlessNode) [reply lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/lib/routine.js:163]: @kgryte I am updating it, should be ready in a few mins.

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/lib/routine.js:163]: Regardless, it is still odd that the actual is the expected value, which makes me believe that the Wasm binary is of a different implementation. @Snehil-Shah Would you mind running `make clean-wasm` on the package and then recompiling?

Comment (Snehil-Shah) [reply lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/lib/routine.js:163]: @kgryte just did, no change..

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/lib/routine.js:163]: @Snehil-Shah Just merged in @headlessNode's changes. Let's recompile and try again. 😅

**Comment by kgryte** [comment]
For reference, the downstream test failures are expected. @headlessNode is currently working on a patch.

Looks like the test failures have now been addressed. Will do a review later today. Thanks for working on this, @Snehil-Shah!

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Snehil-Shah!

**Pull request merged**